### PR TITLE
Fix VIB issues

### PR DESCRIPTION
**Pull request by saifulbkhan**
Minor UX changes along with refactors for algorithmic consistency in peak-detection (specifically manual vs. automatically integrated peaks).

### CODE REVIEW

**Comment by sakshikukreja14** [comment]
While reviewing, I came across a weird bug. When a peak-group is bookmarked, and later a user tries to merge it to a table. A warning appears if he is sure he wants to merge.  Even if clicked "No" to the warning, an alert is generated which says that group is successfully merged. If we can fix it in this branch, it would be great.

File: src/gui/mzroll/settingsform.cpp [hunk 1]
```diff
@@ -221,6 +221,11 @@ SettingsForm::SettingsForm(QSettings* s, MainWindow *w): QDialog(w) {
             &SettingsForm::loadModel);
     _updateModelPath();
 
+    connect(autosaveCheckBox,
```
Comment (sakshikukreja14) [inline src/gui/mzroll/settingsform.cpp:224]: Should we connect _"use default parameters"_ in settings dialog to uncheck auto-save checkbox by default??

Comment (saifulbkhan) [reply src/gui/mzroll/settingsform.cpp:224]: Hm... no. I would want the user to make this decision and then it should stay that way until they chose to change it. "Use default parameters" is as I understand only for algorithm parameters.

**Comment by saifulbkhan** [comment]
@sakshikukreja14 
> A warning appears if he is sure he wants to merge.
This is actually our "delete all peak-groups" warning from the parent peak-table class. When merging, groups are replicated in the other table and deleted from bookmark table - rather than actually moving groups from one to another.

I will make the change such that the warning message does not come up for these merge operations.

**Review by sakshikukreja14 — APPROVED** [review]

**Pull request merged**